tests/pipeline/test_runner.py

```diff
@@ -59,8 +59,11 @@ def test_docker_runner(mock_docker_installation):
         )
 
 
-def test_docker_runner_from_pipeline(mock_docker_installation):
-    with mock.patch("subprocess.call") as mock_call:
+def test_docker_runner_from_pipeline(mock_docker_installation, tmp_path_factory):
+    with mock.patch("subprocess.call") as mock_call, tmp_path_factory.mktemp(
+        "temp",
+    ) as fn:
+        PIPELINE.base_path = str(fn)
```

Comment:
Probably not ideal to set an attribute on a global variable since it creates a side effect? I would either pull it into the test, or make it a pipeline factory function that takes a base path.